### PR TITLE
fix(gui): handle invalid file name edge case on Windows

### PR DESCRIPTION
**Pull request by nilsding**
File names like `"c:blah"` would break the usage of the dialogue, displaying a path like `"c://blah"` instead; whereas file names like `"test:this"` would work.

Apparently `QDir::filePath` checks if the passed file name is an absolute file name using `!QFileInfo::isRelative`, which is the case if the second character in the string is a colon -- probably indicating a drive letter.

### CODE REVIEW

File: src/gui/invalidfilenamedialog.cpp [hunk 1]
```diff
@@ -97,7 +97,7 @@ InvalidFilenameDialog::InvalidFilenameDialog(AccountPtr account,
     switch (invalidMode) {
     case InvalidMode::SystemInvalid:
         _ui->descriptionLabel->setText(tr("The file \"%1\" could not be synced because the name contains characters which are not allowed on this system.").arg(_originalFileName));
-        _ui->explanationLabel->setText(tr("The following characters are not allowed on the system: * \" | & ? , ; : \\ / ~ < > leading/trailing spaces"));
+        _ui->explanationLabel->setText(tr("The following characters are not allowed on the system: \\ / : ? * \"  < > | leading/trailing spaces"));
```
Comment (mgallien) [inline src/gui/invalidfilenamedialog.cpp:100]: I am not sure I understand the change here
what is the purpose ?

File: src/gui/tray/activitylistmodel.cpp [hunk 2]
```diff
@@ -694,8 +694,21 @@ void ActivityListModel::slotTriggerDefaultAction(const int activityIndex)
...
             : InvalidFilenameDialog::InvalidMode::SystemInvalid;
 
+#ifdef Q_OS_WIN
+        // Edge case time!
+        // QDir::filePath checks whether the passed `fileName` is absolute (essentialy by using `!QFileInfo::isRelative()`).
+        // On Windows, if `fileName` starts with a letter followed by a colon (e.g. "A:BCDEF"), it is considered to be an
+        // absolute path.
+        // Since the complete desired path is required by InvalidFilenameDialog, catch that special case here and prefix it ourselves...
+        const auto filePath = activity._file.length() >= 2 && activity._file[1] == ':'
+                              ? folder->path() + activity._file
```
Comment (mgallien) [inline src/gui/tray/activitylistmodel.cpp:704]: why not always using this for all platforms ?
would make the code simpler

**Review by mgallien — CHANGES_REQUESTED** [review]

File: src/gui/invalidfilenamedialog.cpp (hunk 1 above)
Comment (nilsding) [reply src/gui/invalidfilenamedialog.cpp:100]: I was surprised that the description text contained characters that are not part of the `illegalCharacters` array defined above

File: src/gui/tray/activitylistmodel.cpp (hunk 2 above)
Comment (mgallien) [reply src/gui/tray/activitylistmodel.cpp:704]: then maybe that should be made an utility method of `FolderMan::instance()->folder` such that we can ensure we have the same fix everywhere
